docs/aqua-book/language/abilities.md

```diff
@@ -0,0 +1,73 @@
+# Abilities
+
+Abilities are a way to organize code in modules and implement inversion of control pattern in Aqua.
+
+<!-- TODO: Add services here too -->
+Ability is a product of [scalars](types.md#scalars), [structures](types.md#structures), [arrows](types.md#arrow-types) and other abilities.
+
+```aqua
+data Struct:
+    int: i8
+
+ability Simple:
+    st: Struct
+    arrow(x: i8) -> bool
+
+ability Complex:
+    simple: Simple
+    field: string
+```
+
+## Creating abilities
+
+Abilities could be created inside functions just like [structures](types.md#structures).
+
+<!-- TODO: Syntax of creating abilities is a subject to change -->
+```aqua
+func main():
+
+    closure = (x: i8) -> bool:
+        <- x > 0
+
+    MyComplex = Complex(
+        simple = Simple(
+            st = Struct(int = 0),
+            arrow = closure
+        ),
+        field = "complex"
+    )
+```
+
+## Passing abilities
+
+Abilities could be passed to and returned from functions, althrough they require special syntax:
+- Abilities required by function are listed before arguments without names
+- Abilities returned by function are listed in return types just as any other types
+
+```aqua
+ability Additional:
+    value: string
+
+func createComplex{Simple, Additional}(int: i8) -> Complex, string:
+    myComplex = Complex(
```

Comment:
I think we will restrict lower-case variables for abilities and restrict upper-case variables for other types. So, you can be consistent with variable names here